### PR TITLE
Feature/improve appsignal to support tagging

### PR DESCRIPTION
**Pull request by Vuta**
# background

We want to improve Appsignal logging by using tags for metric

# design

- Check if the payload hash is in correct format (needs `value` and `tags` key), else returns failure result.
- Fallback to previous implementation if payload value is not a hash

# test cases

- [ ] `EventTracer.info action: 'Action', message: 'Message', appsignal: { increment_counter: { counter_1: 1, counter_2: 2 } }` works as usual
- [ ] Tags can be added to metric with the format `{ name: { value: value, tags: hash } }`
Example:    
 ```
     EventTracer.info(
       action: 'Action',
       message: 'Message',
       appsignal: {
         increment_counter: {
         counter_1: { value: 1, tags: { region: 'eu' } }
      }})
```
- [ ] Failure returns if tags are in invalid format (missing `value` and `tags`)

### CODE REVIEW

File: lib/event_tracer/appsignal_logger.rb [hunk 1]
```diff
@@ -37,10 +40,22 @@ def applied_metrics(appsignal_args)
...
+          if attribute.is_a?(Hash)
+            begin
+              appsignal.send(
+                metric,
+                increment,
+                attribute.fetch(:value),
+                attribute.fetch(:tags)
+              )
+            rescue KeyError
+              raise Exception, "Appsignal payload { #{increment}: #{attribute} } invalid"
```
Comment (hieuk09) [inline lib/event_tracer/appsignal_logger.rb]: Should we raise a specific EventTracer error instead of `Exception`?

Comment (Vuta) [reply lib/event_tracer/appsignal_logger.rb]: yes I did consider adding a new error class `InvalidTagError` for this one, but somehow decided not to. I just updated it now

**Comment by hieuk09** [comment]
@Vuta could you bump the gem to 0.1.3. The changes look good to me.

File: lib/event_tracer/appsignal_logger.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@
 #
 # Usage: EventTracer.register :appsignal, EventTracer::AppsignalLogger.new(Appsignal)
 #        appsignal_logger.info appsignal: { increment_counter: { counter_1: 1, counter_2: 2 }, set_gauge: { gauge_1: 1 } }
+#        appsignal_logger.info appsignal: { set_gauge: { gauge_1: { value: 1, region: 'eu' } } }
```
Comment (sylv3rblade) [inline lib/event_tracer/appsignal_logger.rb]: just a minor note, we use `attribute.fetch(:tags)` to fetch the tag value but in this example we don't have a `tag` key

Comment (Vuta) [reply lib/event_tracer/appsignal_logger.rb]: sorry for the late response. I was busy with other issues. Just updated the usage, thanks 👍

**Comment by melvrickgoh** [comment]
@Vuta can u help to update this branch to incl. 0.2? I'll help to review the GH logs too

**Review by hieuk09 — APPROVED** [review]
- [x] tag is collected correctly

@Vuta could you bump this gem to 0.2.1?

**Pull request merged**